src/main/python/modelers/regression_models.py

```diff
@@ -0,0 +1,122 @@
+import numpy as np
+import pandas as pd
+from argparse import Namespace
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import LinearRegression
+from sklearn.neural_network import MLPRegressor
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.metrics import r2_score
+from sklearn.metrics import mean_squared_error
+
+
+class TweetsModeler:
+    def __init__(self, args):
+        self.dfWithFeatures = args.dfWithFeatures
+        self.featureColumnIndexes = args.featureColumnIndexes
+        self.points = args.points
+        self.regressionMethod = args.regressionMethod
+        self.regResultsDF = pd.DataFrame()
+        self.resultsDict = dict()
+
+    @classmethod
+    def getModelResutls(cls, dfWithFeatures, featureColumnIndexes, points, regressionMethod):
+        """
+        Returns an tweetsModeler object. the object has two important attributes, regResutlsDF and resultsDict
+
+        @param dfWithFeatures: The data frame that will have the features from an analysis
+        @param featureColumnIndexes: A list of the indexes that contain the feature columns
+        @param points: True if points should be predicted, false if accuracy should be predicted.
+        @param regressionMethod: An integer value signalling which regression method to be used
+        @return: An object with attributes for predictions dataframe and regression results
+        @rtype: object
+        """
+        args = Namespace(
+            dfWithFeatures=dfWithFeatures,
+            featureColumnIndexes=featureColumnIndexes,
+            points=points,
+            regressionMethod=regressionMethod,
+        )
+        tweetsModeler = cls(args)
+        if tweetsModeler.regressionMethod == '1':
+            tweetsModeler.regResultsDF = tweetsModeler.performRegressionAnalysis()
+        elif args.regressionMethod == '2':
+            tweetsModeler.regResultsDF = tweetsModeler.performMLPAnalysis()
+        else:
+            tweetsModeler.regResultsDF = tweetsModeler.performRandomForestAnalysis()
+
+        tweetsModeler.resultsDict = {
+            'r2': r2_score(tweetsModeler.regResultsDF['y_test'], tweetsModeler.regResultsDF['y_pred']),
+            'mse': mean_squared_error(tweetsModeler.regResultsDF['y_test'], tweetsModeler.regResultsDF['y_pred'])
+        }
+        return tweetsModeler
+
+    def __getVariables(self):
```

Comment:
Variables is a little vague to me, maybe change this name to __getTrainTestSplitVars or something like that. getVariables doesn't really tell you what it's doing.